src/systems/ec/implementations/vanilla/packages/ec-component-gallery/ec-component-gallery.js

```diff
@@ -0,0 +1,299 @@
+    this.overlaySelector = overlaySelector;
+    this.overlayHeaderSelector = overlayHeaderSelector;
+    this.overlayFooterSelector = overlayFooterSelector;
+    this.overlayImageSelector = overlayImageSelector;
+    this.overlayCounterCurrentSelector = overlayCounterCurrentSelector;
+    this.overlayCounterMaxSelector = overlayCounterMaxSelector;
+    this.overlayDescriptionSelector = overlayDescriptionSelector;
+    this.overlayMetaSelector = overlayMetaSelector;
+    this.overlayPreviousSelector = overlayPreviousSelector;
+    this.overlayNextSelector = overlayNextSelector;
+    this.attachClickListener = attachClickListener;
+    this.attachKeyListener = attachKeyListener;
+
+    // Private variables
+    this.galleryItems = null;
+    this.meta = null;
+    this.closeButton = null;
+    this.overlay = null;
+    this.overlayHeader = null;
+    this.overlayFooter = null;
+    this.overlayImage = null;
+    this.overlayCounterCurrent = null;
+    this.overlayCounterMax = null;
+    this.overlayDescription = null;
+    this.overlayMeta = null;
+    this.overlayPrevious = null;
+    this.overlayNext = null;
+    this.selectedItem = null;
+
+    // Bind `this` for use in callbacks
+    this.handleClickOnCloseButton = this.handleClickOnCloseButton.bind(this);
+    this.handleClickOnItem = this.handleClickOnItem.bind(this);
+    this.handleKeyPressOnItem = this.handleKeyPressOnItem.bind(this);
+    this.handleClickOnPreviousButton = this.handleClickOnPreviousButton.bind(
+      this
+    );
+    this.handleClickOnNextButton = this.handleClickOnNextButton.bind(this);
+    this.handleKeyboard = this.handleKeyboard.bind(this);
+  }
+
+  init() {
+    // Query elements
+    this.galleryItems = queryAll(this.galleryItemSelector, this.element);
+    this.closeButton = queryOne(this.closeButtonSelector, this.element);
+    this.overlay = queryOne(this.overlaySelector, this.element);
+    this.overlayHeader = queryOne(this.overlayHeaderSelector, this.overlay);
+    this.overlayFooter = queryOne(this.overlayFooterSelector, this.overlay);
+    this.overlayImage = queryOne(this.overlayImageSelector, this.overlay);
+    this.overlayCounterCurrent = queryOne(
+      this.overlayCounterCurrentSelector,
+      this.overlay
+    );
+    this.overlayCounterMax = queryOne(
+      this.overlayCounterMaxSelector,
+      this.overlay
+    );
+    this.overlayDescription = queryOne(
+      this.overlayDescriptionSelector,
+      this.overlay
+    );
+    this.overlayMeta = queryOne(this.overlayMetaSelector, this.overlay);
+    this.overlayPrevious = queryOne(this.overlayPreviousSelector, this.overlay);
+    this.overlayNext = queryOne(this.overlayNextSelector, this.overlay);
+
+    // Polyfill to support <dialog>
+    this.isDialogSupported = true;
+    if (!window.HTMLDialogElement) {
+      this.isDialogSupported = false;
+    }
+
+    // Hack to detect when focus gets out of the modal
+    this.overlay.addEventListener('transitionend', () => {
```

Comment:
This hack doesn't work well. If you click anywhere on the modal, the focus will always come back to the close button. 

Have you considered other options like https://github.com/davidtheclark/focus-trap or https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element or https://allyjs.io/tutorials/accessible-dialog.html?